### PR TITLE
[6.3] Create VMware compute resource without url (BZ:1387917)

### PR DESCRIPTION
**Pull request by ldjebran**
cover https://bugzilla.redhat.com/show_bug.cgi?id=1387917
```shell
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_computeresource_vmware.py
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 2 items 
2017-11-07 15:12:41 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/cli/test_computeresource_vmware.py::VMWareComputeResourceTestCase::test_positive_create_with_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_computeresource_vmware.py::VMWareComputeResourceTestCase::test_positive_create_with_server <- robottelo/decorators/__init__.py PASSED

============================================== 2 passed in 33.47 seconds ===============================================
```

### CODE REVIEW

File: robottelo/cli/factory.py [hunk 1]
```diff
@@ -1840,6 +1840,7 @@ def make_compute_resource(options=None):
 
     Options::
 
+        --datacenter DATACENTER       for RHEV, VMware Datacenter
```
Comment (oshtaier) [inline robottelo/cli/factory.py]: Update docstring and parameters to actual state:
```
def make_compute_resource(options=None):
    """
    Usage::

        hammer compute-resource create [OPTIONS]

    Options::

        --caching-enabled CACHING_ENABLED    enable caching, for VMware only
                                             One of true/false, yes/no, 1/0.
        --datacenter DATACENTER       for RHEV, VMware Datacenter
        --description DESCRIPTION
        --display-type DISPLAY_TYPE   for Libvirt only
                                      Possible value(s): 'VNC', 'SPICE'
        --location-ids LOCATION_IDS   Comma separated list of values.
        --location-titles LOCATION_TITLES    Comma separated list of values.
        --locations LOCATION_NAMES    Comma separated list of values.
        --name NAME
        --organization-ids ORGANIZATION_IDS  Comma separated list of values.
        --organization-titles ORGANIZATION_TITLES    Comma separated list of
                                                     values.
        --organizations ORGANIZATION_NAMES   Comma separated list of values.
        --password PASSWORD           Password for Ovirt, EC2, Vmware,
                                      Openstack. Secret key for EC2
        --provider PROVIDER           Providers include Libvirt, Ovirt, EC2,
                                      Vmware, Openstack, Rackspace, GCE
        --region REGION               for EC2 only
        --server SERVER               for Vmware
        --set-console-password SET_CONSOLE_PASSWORD for Libvirt and Vmware only
                                                    One of true/false,
                                                    yes/no, 1/0.
        --tenant TENANT               for Openstack only
        --url URL                     URL for Libvirt, Ovirt, and Openstack
        --user USER                   Username for Ovirt, EC2, Vmware,
                                      Openstack. Secret key for EC2
        --uuid UUID                   for Ovirt, Vmware Datacenter
        -h, --help                    print help

    """
    args = {
        u'caching-enabled': None,
        u'datacenter': None,
        u'description': None,
        u'location-ids': None,
        u'location-titles': None,
        u'locations': None,
        u'name': gen_alphanumeric(),
        u'organization-ids': None,
        u'organization-titles': None,
        u'organizations': None,
        u'password': None,
        u'provider': None,
        u'region': None,
        u'server': None,
        u'set-console-password': None,
        u'tenant': None,
        u'url': None,
        u'user': None,
        u'uuid': None,
    }
```

Comment (oshtaier) [reply robottelo/cli/factory.py]: 2)
I would suggest to remove:
```
    #
    # if options is None:
    #     options = {}

    # if options.get('provider') is None:
    #     options['provider'] = FOREMAN_PROVIDERS['libvirt']
        # if options.get('url') is None:
        #     options['url'] = 'qemu+tcp://localhost:16509/system'
```
with corresponding re-test for all cases involved.

File: tests/foreman/cli/test_computeresource_vmware.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            'server': self.vmware_server,
+            'user': self.vmware_username,
+            'password': self.vmware_password,
+            'datacenter': self.vmware_datacenter
+        })
+        self.assertEquals(vmware_cr['name'], cr_name)
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_create_with_server(self):
```
Comment (oshtaier) [inline tests/foreman/cli/test_computeresource_vmware.py]: I would say that second test is duplication to the first one or at least second one should focus on organization and location which is actual difference to first case.

Comment (ldjebran) [reply tests/foreman/cli/test_computeresource_vmware.py]: @oshtaier that was a problem as I have to make sure that the server field is saved correctly, and as we have no tests vmware cr for cli we have to make sure that a test with name exist, so the first test is a simple create with name to make sure that we can create a vmware cr,  the second one with org and loc and assert on server field saved

Comment (oshtaier) [reply tests/foreman/cli/test_computeresource_vmware.py]: Yeah, I can read the code=) So just make three cases ('test_positive_create_with_server`), (`test_positive_create_with_org`) and (`test_positive_create_with_server`). And check server + name in the first

File: robottelo/cli/factory.py (hunk 1 above)
Comment (ldjebran) [reply robottelo/cli/factory.py]: @oshtaier, think this is not directly related to the current tests, 
there is too many tests involved in different test modules , can be done in a  separate PR (that  can be also blocked by other reviewer)
of course will update that create usage help doc string

**Comment by ldjebran** [comment]
```bash
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_computeresource_vmware.py
=================================================================================================== test session starts ====================================================================================================
platform linux2 -- Python 2.7.13, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 3 items                                                                                                                                                                                                           
2017-11-09 18:31:23 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/cli/test_computeresource_vmware.py::VMWareComputeResourceTestCase::test_positive_create_with_loc <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_computeresource_vmware.py::VMWareComputeResourceTestCase::test_positive_create_with_org <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_computeresource_vmware.py::VMWareComputeResourceTestCase::test_positive_create_with_server <- robottelo/decorators/__init__.py PASSED

================================================================================================ 3 passed in 43.11 seconds =================================================================================================
```

**Comment by ldjebran** [comment]
@oshtaier comments addressed

**Comment by ldjebran** [comment]
```
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ pytest -v tests/foreman/cli/test_computeresource_vmware.py
=================================================================================================== test session starts ====================================================================================================
platform linux2 -- Python 2.7.13, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 4 items                                                                                                                                                                                                           
2017-11-09 18:56:34 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/cli/test_computeresource_vmware.py::VMWareComputeResourceTestCase::test_positive_create_with_loc <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_computeresource_vmware.py::VMWareComputeResourceTestCase::test_positive_create_with_org <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_computeresource_vmware.py::VMWareComputeResourceTestCase::test_positive_create_with_org_and_loc <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_computeresource_vmware.py::VMWareComputeResourceTestCase::test_positive_create_with_server <- robottelo/decorators/__init__.py PASSED

================================================================================================ 4 passed in 50.76 seconds =================================================================================================
```

**Pull request merged**